### DISCUSSION TITLE
SupaBase Public URL - How to show all users avatar?

### DISCUSSION THREAD

**Discussion by charlesoestergaard** [supabase/supabase#5363] [Questions] [answered]
**Framework: Next.js with SupaBase**

Im trying to get all users avatar that are stored in a column in a table -> "Profiles"

[![SupaBase Table][1]][1]

I only get the image name "placeholder.png", how do i get the the public URL, and map over it for each user ?

This is my code

````javascript

    
    function Brugere({ profiler, url }) {
      return (
        <div className="flex -space-x-2 overflow-hidden">
          {console.log(profiler)}
          {profiler.map((e, index) => (
            <div className="person" key={index}>
              <p>{e.username}</p>
    
              <img
                className="inline-block h-10 w-10 rounded-full ring-2 ring-white"
                src={e.avatar_url}
                alt=""
              />
            </div>
          ))}
        </div>
      );
    }
    
    export default Brugere;
    
    export async function getServerSideProps() {
      const { data: profiler, error } = await supabase.from("profiles").select("*");
    
      {
        error ? console.log(error) : console.log(profiler);
      }
    
      return {
        props: {
          profiler,
        },
      };
    }
````

Result:

[![Final Result][2]][2]

  [1]: https://i.stack.imgur.com/wGgrO.png
  [2]: https://i.stack.imgur.com/4DfWa.png

**Comment by GaryAustin1** [comment, marked as answer]
If it is in a public bucket then the URL is somewhat fixed.  Storage getPublicURL just builds up the path from constants and bucket/path/filename and does not make a storage call.

Format is:
https://YourSupabaseURL/storage/v1/object/public/bucket/pathtofilename
so https://xkindfasdfsvnhhhhjjjjmvblxddd.supabase.co/storage/v1/object/public/test/pred.gif  where test is bucket and pred.gif is filename (note it could also be a pathname to the file if not at first level in bucket.).

So just append a string to your filename...

Edit: You can get also get  the string from supabase.storageUrl and it will have the url up to and including "/v1".

**Reply by charlesoestergaard** [reply]
Thanks so much for you time to respond.

I tried to do as you said, but without any succses. 

How i inserted it -> 

```jsx
          <img
            className="inline-block h-10 w-10 rounded-full ring-2 ring-white"
            src={
              "https://obhvrqaxovhnpboqxtxq.supabase.in/storage/v1/object/public/avatars/ " +
              e.avatar_url
            }
            alt=""
          />
          
```

Final URL: https://obhvrqaxovhnpboqxtxq.supabase.in/storage/v1/object/public/avatars/20955289431.jpg

I dont understand this supabase.storageUrl ? How would i use it? The problem is that theres mabye 1000+ users, so i need it to be done in a loop and not manually for each user. 😊

**Reply by GaryAustin1** [reply]
You have an extra space at end of your constant string... avatars/ "

const storageUrl = supabase.storage.Url after client init.  Maybe was not clear that is just your base address.

Then use storageUrl + "/object/public/avatars/" + e.avatar_url

edit: Or avatarUrl = supabase.storageUrl + /object/public/avatars/" then avatarUrl + e.avatar_url

**Reply by charlesoestergaard** [reply]
I noticed the space, but as the final url dident have any i guessed it dident mattered.


Now i have this, and still doesnt work.

https://i.imgur.com/UWryo1f.png

**Reply by GaryAustin1** [reply]
supabase.storageUrl  with capital and no period.
You can see it if you console.log (supabase)

**Reply by charlesoestergaard** [reply]
I think you misunderstand. 

It works as intended, i get the right url for all images. But it doesn't show the image.


It doesnt work with capital Url (supabase.storage.Url), but works with url (supabase.storage.url).

I get the error (Resource not found) even though its the correct image.

https://obhvrqaxovhnpboqxtxq.supabase.co/storage/v1/object/public/avatars/955289431.jpg

**Reply by GaryAustin1** [reply]
Is it a public bucket?

**Reply by charlesoestergaard** [reply]
How can i tell? 🤷‍♂️

**Reply by GaryAustin1** [reply]
![image](https://user-images.githubusercontent.com/54564956/152888841-bf38d93c-0e72-4161-9d23-5906de409930.png)

**Reply by charlesoestergaard** [reply]
Omg its not public.... 🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️


How can i make it public?

**Reply by GaryAustin1** [reply]
Hover over the bucket name and you will see ...  well vertically three dots select public.

**Reply by charlesoestergaard** [reply]
@GaryAustin1 Thank you very much sir, im an idi**. Now everything works and all profiles images are shown! Awesome!
